### DISCUSSION TITLE
Runtime switching Server/Client Authority on NetworkTransform

### DISCUSSION THREAD

**Discussion by TeamSailEnthusiast** [FirstGearGames/FishNet#492] [Feature Request]
Would useful for a lot of features.
Server validated authority is on top of that list:
Client authority is on by default > Server validates > Server catches invalid movement/speed/whatever > switch to Server Auth for a few frames == place back into valid movement range.speed/whatever > give back client auth. 
This way the client will always have butter smooth movement, no input lagg, no need for prediction.

**Comment by FirstGearGames** [comment]
This already exist to a degree. There are events inside the NT you can subscribe to for validating movement and you can kick the client if you don't agree, or reject the movement and more.
